File: Database/ServerWebApi/Model/ContainerItem.cs
namespace ContainerApi.Models;

using Ingredient.Model;
public class ContainerItem
{
    /*
    *@brief This function creates the get and set functions for ContainerItemId
    */
    public int ContainerItemId { get; set; }

    /*
    *@brief This function creates the get and set functions for Place
    */
    public int Place { get; set; }


    /*
    *@brief This function creates the get and set functions for Ingredient
    */
    public IngredientItem Ingredient { get; set; } = default!;


    //public List<Ingredient>[] ingredient { get; set; } = default!;

}

File: Client/Test_of_client_niv/Containeritem.cpp
#include "Containeritem.h"

ContainerItem::ContainerItem(int containerItemId, int place)
{
    setContainerId(containerItemId);
    setPlace(place);
}

int ContainerItem::getContainerId() const
{
    return _containerItemId;
}

void ContainerItem::setContainerId(int containerItemId)
{
    _containerItemId = containerItemId;
}

int ContainerItem::getPlace() const
{
    return _place;
}

void ContainerItem::setPlace(int place)
{
    _place = place;
}

void ContainerItem::operator=(const ContainerItem &temp){
    _containerItemId = temp.getContainerId();
    _place = temp.getPlace();
    _ingredient = temp.getIngredient();

}

void ContainerItem::setIngredient(const IngredientItem& temp){
    _ingredient = temp;
}

IngredientItem ContainerItem::getIngredient() const{
    return _ingredient;
}

ContainerItem::~ContainerItem(){

}
